grammar.js

```diff
@@ -1000,29 +1005,83 @@ module.exports = grammar({
       seq('[', commaSep($.array_element_initializer), optional(','), ']')
     ),
 
-    string: $ => {
-      const b_prefix = /[bB]/
-      const single_quote_chars = repeat(/\\'|\\\\|\\?[^'\\]/)
-      const single_quote_string = seq(
-        optional(b_prefix), "'", single_quote_chars, "'"
+    _string_member_access_expression: $ => prec(PREC.MEMBER, seq(
+      field('object', $._string_dereferencable_expression),
+      '->',
+      $._member_name
+    )),
+
+    _string_dereferencable_expression: $ => prec(PREC.DEREF, choice(
+      $._variable_name,
+      alias($._string_subscript_expression, $.subscript_expression),
+      alias($._string_member_access_expression, $.member_access_expression)
+    )),
+
+    _string_subscript_expression: $ => seq(
+      $._string_dereferencable_expression,
+      choice(
+        seq('[', optional($._expression), ']'),
+        seq('{', $._expression, '}')
       )
+    ),
+
+    _complex_string_part: $ => prec.right(seq(
+      "{",
+      choice(
+        alias($._string_member_access_expression, $.member_access_expression),
+        $._variable_name,
+        alias($._string_subscript_expression, $.subscript_expression),
+      ),
+      "}"
+    )),
 
-      const double_quote_string = seq(
-        optional(b_prefix), '"', double_quote_chars(), '"'
+    _simple_string_part: $ => choice(
+      alias($._string_member_access_expression, $.member_access_expression),
+      $._variable_name,
```

Comment:
If I remove `$._variable_name`, the two other rules comes to effect (and variables does not work for obvious reasons), but for some reason eluding me, the two other rules are never invoked if this line is available.